pyro/contrib/gp/kernels/kernel.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import absolute_import, division, print_function
+
+import torch.nn as nn
+
+
+class Kernel(nn.Module):
+    """
+    Base class for kernels used in Gaussian Process.
+
+    Every inherited class should implement the forward pass which
+        take inputs X, X2 and return their covariance matrix.
+    """
+
+    def __init__(self, active_dims=None, name=None):
+        super(Kernel, self).__init__()
+        self.active_dims = active_dims
+        self.name = name
+
+    def K(self, X, Z=None):
```

Comment:
nit: Do you think this method would be better named `__call__`?

Reply:
Yes, it is unnecessary right now. So I remove it. From now on, each forward method in derived kernels should include `_slice_X` statements.